src/Speech/V1beta1/SpeechClient.php

```diff
@@ -0,0 +1,369 @@
+ * limitations under the License.
+ */
+
+/*
+ * GENERATED CODE WARNING
+ * This file was generated from the file
+ * https://github.com/google/googleapis/blob/master/google/cloud/speech/v1beta1/cloud_speech.proto
+ * and updates to that file get reflected here through a refresh process.
+ *
+ * EXPERIMENTAL: this client library class has not yet been declared beta. This class may change
+ * more frequently than those which have been declared beta or 1.0, including changes which break
+ * backwards compatibility.
+ */
+
+namespace Google\Cloud\Speech\V1beta1;
+
+use Google\GAX\AgentHeaderDescriptor;
+use Google\GAX\ApiCallable;
+use Google\GAX\CallSettings;
+use Google\GAX\GrpcConstants;
+use Google\GAX\GrpcCredentialsHelper;
+use Google\GAX\LongRunning\OperationsClient;
+use google\cloud\speech\v1beta1\AsyncRecognizeRequest;
+use google\cloud\speech\v1beta1\AsyncRecognizeResponse;
+use google\cloud\speech\v1beta1\RecognitionAudio;
+use google\cloud\speech\v1beta1\RecognitionConfig;
+use google\cloud\speech\v1beta1\SpeechGrpcClient;
+use google\cloud\speech\v1beta1\SyncRecognizeRequest;
+
+/**
+ * Service Description: Service that implements Google Cloud Speech API.
+ *
+ * EXPERIMENTAL: this client library class has not yet been declared beta. This class may change
+ * more frequently than those which have been declared beta or 1.0, including changes which break
+ * backwards compatibility.
+ *
+ * This class provides the ability to make remote calls to the backing service through method
+ * calls that map to API methods. Sample code to get started:
+ *
+ * ```
+ * try {
+ *     $speechClient = new SpeechClient();
+ *     $config = new RecognitionConfig();
+ *     $audio = new RecognitionAudio();
+ *     $response = $speechClient->syncRecognize($config, $audio);
+ * } finally {
+ *     if (isset($speechClient)) {
+ *         $speechClient->close();
+ *     }
+ * }
+ * ```
+ *
+ * Many parameters require resource names to be formatted in a particular way. To assist
+ * with these names, this class includes a format method for each type of name, and additionally
+ * a parse method to extract the individual identifiers contained within names that are
+ * returned.
+ */
+class SpeechClient
+{
+    /**
+     * The default address of the service.
+     */
+    const SERVICE_ADDRESS = 'speech.googleapis.com';
+    /**
+     * The default port of the service.
+     */
+    const DEFAULT_SERVICE_PORT = 443;
+
+    /**
+     * The default timeout for non-retrying methods.
+     */
+    const DEFAULT_TIMEOUT_MILLIS = 30000;
+
+    const _CODEGEN_NAME = 'gapic';
+    const _CODEGEN_VERSION = '0.1.0';
+
+    private $grpcCredentialsHelper;
+    private $speechStub;
+    private $scopes;
+    private $defaultCallSettings;
+    private $descriptors;
+    private $operationsClient;
+
+    private static function getPageStreamingDescriptors()
+    {
+        $pageStreamingDescriptors = [
+        ];
+
+        return $pageStreamingDescriptors;
+    }
```

Comment:
Can this logic be omitted (as well as the logic that calls this function below) when we know the page streaming descriptors are empty?